plugin/template/template.go

```diff
@@ -128,6 +142,29 @@ func (h Handler) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg)
 // Name implements the plugin.Handler interface.
 func (h Handler) Name() string { return "template" }
 
+func executeEdnsTemplate(server, section string, template *gotmpl.Template, o *dns.OPT, data *templateData) error {
+	buffer := &bytes.Buffer{}
+	err := template.Execute(buffer, data)
+	if err != nil {
+		templateFailureCount.WithLabelValues(server, data.Zone, data.Class, data.Type, section, template.Tree.Root.String()).Inc()
+		return err
+	}
+	pos := strings.Index(buffer.String(), "=")
+	i, err := strconv.Atoi(buffer.String()[0:pos])
+	if err != nil {
+		templateRRFailureCount.WithLabelValues(server, data.Zone, data.Class, data.Type, section, template.Tree.Root.String()).Inc()
+		return err
+	}
+	code := uint16(i)
+	var edata []byte
+	if buffer.Len() > pos {
+		edata = buffer.Bytes()[pos+1:]
```

Comment:
In `metadata_edns0` we allow decoding hex, addresses, and bytes. Wouldn't it make sense to allow the encodings of that here, especially given that data may come from template data (which IIRC could be metadata added by another plugin).

https://github.com/coredns/metadata_edns0

What I am getting at is, shouldn't we be able to encode any binary data in the value?

Reply:
I was assuming we could do this with escapes .... like "\x1f" ...
I'll check to see if that actualy works.

Reply:
darn that doesn't work.  Plain strings is all I need for my tests, so i'll just keep this around for my own personal use.